core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -321,6 +321,20 @@ private static String toSql(RelNode root, SqlDialect dialect) {
         .ok(expectedMySql);
   }
 
+  @Test public void testSelectQueryWithGroupBySubQuery() {
+    String query = "select \"product_class_id\", count(\"product_id\")\n"
+        + "from (select \"product_class_id\", \"product_id\"\n"
+        + "from \"product\"\n"
+        + "group by \"product_class_id\", \"product_id\") as t\n"
+        + "group by \"product_class_id\"";
+    final String expected = "SELECT \"product_class_id\", COUNT(*)\n"
```

Comment:
This is incorrect. `COUNT(*)` doesn't equal with `count("product_id")`, if product_id is nullable.

Reply:
Surprisingly this test passes.

Reply:
It's indeed another bug, I will make another issue to discuss it.

Reply:
@hsyuan @amaliujia You can see this issue.

Reply:
@hsyuan @amaliujia This my not be a bug, you can see the explaining in https://issues.apache.org/jira/browse/CALCITE-3543

Reply:
Agreed. Either we what improve CALCITE-3543 or not, it shouldn't block this PR.